packages/react-reconciler/src/__tests__/ReactSuspenseList-test.js

```diff
@@ -3119,4 +3119,197 @@ describe('ReactSuspenseList', () => {
       );
     },
   );
+
+  // @gate enableSuspenseList && enableAsyncIterableChildren
+  it('warns for async generator components in "forwards" order', async () => {
+    async function* Generator() {
+      yield 'A';
+      yield 'B';
+    }
+    function Foo() {
+      return (
+        <SuspenseList revealOrder="forwards">
+          <Generator />
```

Comment:
This pattern would be very convenient but we can't support it since we need to resolve the rows first.

Interestingly this pattern is supported in Server Components because they get resolved into the iterable on the server and the component wrapper disappears. It's just not supported in Client Components.